src/util/opening_hours.cpp

```diff
@@ -218,9 +218,9 @@ struct opening_hours_grammar : qi::grammar<Iterator, Skipper, std::vector<Openin
         weekday_sequence = (weekday_range % ',')[ph::bind(&OpeningHours::weekdays, _r1) = _1];
 
         weekday_range
-            = wday[_a = _1, _b = _1]
-            >> -(('-' >> wday[_b = _1])
-                 | ('[' >> (nth_entry % ',') >> ']' >> -day_offset))
+            = (wday[_a = _1, _b = _1]
```

Comment:
@MoKob i can explain what happened in this case: without parenthesis the semantic action `[_val = ph::construct<OpeningHours::WeekdayRange>(_a, _b)]` that constructs  a `WeekdayRange` was invoked only in a case if both the first and the second part of the rule succeed.

In case `Mo-Fr` local rule variable `_a` will be `Mo` and `_b` will be `Fr`. So the actions will be called as `ph::construct<OpeningHours::WeekdayRange>(1, 5)`.

If parsing of the second rule failed but the first part succeeded then  both `_a` and `_b` will be initialized to let say `Sa`, but construction of a `ph::construct<OpeningHours::WeekdayRange>(6, 6)` will not be called. So `WeekdayRange` will be a default initialized object with an empty range.

Adding parenthesis around both rules makes invocation of `ph::construct<OpeningHours::WeekdayRange>(_a, _b)` in both cases: 
- if the first  part succeeds, like 'Mo' -> `ph::construct<OpeningHours::WeekdayRange>(1,1)`
- if the first   and the second optional part  succeeds, like 'Tu-Fr' -> `ph::construct<OpeningHours::WeekdayRange>(2,5)`

Idk if such lengthy comments would make sense, but I can make a step-by-step intro how to add semantic to unimplemented  parts like `nth_entry` or `day_offset`, For example, with `nth_entry` it is possible to define `Oct Su[1]` (the first Sunday in October) like in the Oktoberfest example.